src/Shared/BrowserTesting/src/ContextInformation.cs

```diff
@@ -55,10 +55,24 @@ internal BrowserContextOptions ConfigureUniqueHarPath(BrowserContextOptions brow
             if (browserContextOptions?.RecordHar?.Path != null)
             {
                 var identifier = Guid.NewGuid().ToString("N");
-                browserContextOptions.RecordHar.Path = Path.Combine(browserContextOptions.RecordHar.Path, $"{identifier}.har");
+                var harDirectory = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
+                if (string.IsNullOrEmpty(harDirectory))
+                {
+                    harDirectory = browserContextOptions.RecordHar.Path;
+                }
+                browserContextOptions.RecordHar.Path = Path.Combine(harDirectory, $"{identifier}.har");
                 _harPath = browserContextOptions.RecordHar.Path;
             }
 
+            if (browserContextOptions?.RecordVideo?.Dir != null)
+            {
+                var uploadDir = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
```

Comment:
nit: This could be done just once above [line 55](https://github.com/dotnet/aspnetcore/pull/30637/files#diff-17650b1a61777da7cbc98537ba3541fcb1d0d79fb02cc5af926cde1d3789fee9R55)

Reply:
Yup, will fix in next pr